docs/cluster_spec.md

```diff
@@ -159,9 +158,39 @@ Will result in the flag `--feature-gates=ExperimentalCriticalPodAnnotation=true,
 
 ### networkID
 
-On AWS, this is the id of the VPC the cluster is created in. If creating a cluster from scratch, this field doesn't need to be specified at create time; `kops` will create a `VPC` for you.
+On AWS, this is the id of the VPC the cluster is created in. If creating a cluster from scratch, this field does not need to be specified at create time; `kops` will create a `VPC` for you.
 
 ```yaml
 spec:
   networkID: vpc-abcdefg1
 ```
+
+More information about running in an existing VPC is [here](run_in_existing_vpc.md).
+
+### hooks
+
+Hooks allow the execution of a container before the installation of Kubneretes on every node in a cluster.  For intance you can install nvidia drivers for using GPUs.
+
+```
+spec:
+  # many sections removed
+  hooks:
+  - execContainer:
+      image: kopeio/nvidia-bootstrap:1.6
+```
+
+Install Ceph
+
+```
+spec:
+  # many sections removed
+  hooks:
+  - execContainer:
+      command:
+      - sh
+      - -c
+      - chroot /rootfs apt-get update && chroot /rootfs apt-get install -y ceph-common
+      image: busybox
+```
+
```

Comment:
correct me if I'm wrong. 
I came from the issue #2230 because there is a link directing to this hook document
Does this means the command will be executed on the node during the launching?
And will the image be used as the OS for both master and node instances?

I think it's a general situation that after the instance has been setup, we need to install an agent for each instance within the cluster to invoke the third-party security solution e.g. IPS